mteb/models/jina_models.py

```diff
@@ -213,72 +216,327 @@ def encode(
         return embeddings
 
 
-class JinaV4Wrapper(SentenceTransformerWrapper):
+class JinaV4Wrapper(Wrapper):
     """following the hf model card documentation."""
 
-    jina_task_to_prompt = {
-        "retrieval.query": "Query: ",
-        "retrieval.passage": "Passage: ",
-        "text-matching": "Query: ",
-    }
+    SUPPORTED_VECTOR_TYPES = {"single_vector", "multi_vector"}
 
     def __init__(
         self,
         model: str,
         revision: str | None = None,
+        device_map="cuda",
+        torch_dtype=torch.bfloat16,
+        attn_implementation="flash_attention_2",
+        trust_remote_code: bool = True,
         model_prompts: dict[str, str] | None = None,
+        vector_type: str = "multi_vector",
         **kwargs,
     ) -> None:
+        if CURRENT_SENTENCE_TRANSFORMERS_VERSION < MIN_SENTENCE_TRANSFORMERS_VERSION:
+            raise RuntimeError(
+                f"sentence_transformers version {st_version} is lower than the required version 3.1.0"
+            )
```

Comment:
If Jinav4 is loaded from transformers, maybe we don't need to check sentence transformers version?